api/afdb/userName.ts

```diff
@@ -0,0 +1,23 @@
+"use server"
+
+import axios from 'axios';
+import getAxiosConfig from '../axiosConfig';
+
+export async function getUserName(studentId: string): Promise<string> {
+    const url = process.env.AF_DB_SERVICE_URL;
+    const bearerToken = process.env.AF_DB_SERVICE_BEARER_TOKEN || '';
+
+    try {
+        const response = await axios.get(`${url}/student`, {
+            params: { student_id: studentId },
+            ...getAxiosConfig(bearerToken),
+        });
+
+        const firstName = response.data[0].user.first_name;
```

Comment:
what data does this request return? Maybe return the full response now itself or save it globally somewhere. We will probably use user details for other stuff

Reply:
It will return the username of the student. Once we get the username, it is saved in the state variable so I don't feel saving it globally is a good approach 🤔

Reply:
I think what Pritam is saying is, save the whole user object as a state variable, that way when the user object contains other attributes, you won't have to extract each from the response and save it to state.

Reply:
Yes, what @sudheshna-donthineni is saying

Reply:
oh cool